src/main.rs

```diff
@@ -52,117 +51,66 @@ impl GlobalConfig {
-                                .short('b')
-                                .long("baseline")
-                                .value_name("BASELINE_RUSTDOC_JSON")
-                                .help("The rustdoc json file to use as a semver baseline. Required.")
-                                .action(ArgAction::Set)
-                                .value_parser(value_parser!(PathBuf))
-                                .required(true)
-                        )
-                )
-                .subcommand(
-                    Command::new("check-release")
-                        .arg_required_else_help(true)
-                        .arg(
-                            Arg::new("current_rustdoc_path")
-                                .short('c')
-                                .long("current")
-                                .value_name("CURRENT_RUSTDOC_JSON")
-                                .help("The current rustdoc json output to test for semver violations. Required.")
-                                .action(ArgAction::Set)
-                                .value_parser(value_parser!(PathBuf))
-                                .required(true)
-                        )
-                        .arg(
-                            Arg::new("baseline_rustdoc_path")
-                                .short('b')
-                                .long("baseline")
-                                .value_name("BASELINE_RUSTDOC_JSON")
-                                .help("The rustdoc json file to use as a semver baseline. Required.")
-                                .action(ArgAction::Set)
-                                .value_parser(value_parser!(PathBuf))
-                                .required(true)
-                        )
-                )
-        )
+#[derive(Parser)]
+#[clap(name = "cargo")]
+#[clap(bin_name = "cargo")]
+#[clap(version, propagate_version = true)]
+enum Cargo {
+    #[clap(subcommand)]
+    SemverChecks(SemverChecks),
+}
+
+/// Check your crate for semver violations.
+#[derive(Subcommand)]
+enum SemverChecks {
+    DiffFiles(DiffFiles),
+    CheckRelease(CheckRelease),
+}
+
+#[derive(Args)]
+struct DiffFiles {
+    /// The current rustdoc json output to test for semver violations. Required.
+    #[clap(short, long = "current", value_name = "CURRENT_RUSTDOC_JSON")]
+    current_rustdoc_path: PathBuf,
+
+    /// The rustdoc json file to use as a semver baseline. Required.
+    #[clap(short, long = "baseline", value_name = "BASELINE_RUSTDOC_JSON")]
+    baseline_rustdoc_path: PathBuf,
+}
+
+#[derive(Args)]
+struct CheckRelease {
+    /// The current rustdoc json output to test for semver violations. Required.
+    #[clap(short, long = "current", value_name = "CURRENT_RUSTDOC_JSON")]
+    current_rustdoc_path: PathBuf,
+
+    /// The rustdoc json file to use as a semver baseline. Required.
+    #[clap(short, long = "baseline", value_name = "BASELINE_RUSTDOC_JSON")]
+    baseline_rustdoc_path: PathBuf,
```

Comment:
Since these two subcommands are identical (though poorly documented as such), is there a way to have them use an identical struct instead of two different ones?

Reply:
https://github.com/obi1kenobi/cargo-semver-check/pull/78 is the follow up to this comment